vault/resource_aws_secret_backend_role.go

```diff
@@ -130,25 +130,15 @@ func awsSecretBackendRoleWrite(d *schema.ResourceData, meta interface{}) error {
 	data := map[string]interface{}{
 		"credential_type": credentialType,
 	}
-	if policy != "" {
-		data["policy_document"] = policy
-	}
-	if len(policyARNs) != 0 {
-		data["policy_arns"] = policyARNs
-	}
-	if len(roleARNs) != 0 {
-		data["role_arns"] = roleARNs
-	}
+	data["policy_document"] = strings.TrimSpace(policy.(string))
+	data["policy_arns"] = policyARNs
+	data["role_arns"] = roleARNs
 
 	defaultStsTTL, defaultStsTTLOk := d.GetOk("default_sts_ttl")
 	maxStsTTL, maxStsTTLOk := d.GetOk("max_sts_ttl")
 	if credentialType == "assumed_role" || credentialType == "federation_token" {
-		if defaultStsTTLOk {
-			data["default_sts_ttl"] = strconv.Itoa(defaultStsTTL.(int))
-		}
-		if maxStsTTLOk {
-			data["max_sts_ttl"] = strconv.Itoa(maxStsTTL.(int))
-		}
+		data["default_sts_ttl"] = strconv.Itoa(defaultStsTTL.(int))
+		data["max_sts_ttl"] = strconv.Itoa(maxStsTTL.(int))
```

Comment:
For these (line 140 and 141), should we only set them if they've been sent by the caller? 

What I'm thinking about is, what if they send nothing, and the code here resolves to `data["max_sts_ttl"] = "0"`. Then, Vault gets the call and sees the `max_sts_ttl` key _included_ and views it as an intentional 0, rather than as _not included_. I'm not sure what Vault's behavior would be in that situation, and I'm not sure it will always behave the same way in the future as it does today.

Reply:
Unfortunately, I did not find an endpoint that allows us to overwrite the whole role instead of only updating it, so it's a hard call: either we follow the Vault behavior (keep previous values), or we favor the Terraform principle where given a configuration we need a stable output.

For integer values, the "default" value being 0 stays consistent with go's default value, which is why I believe this won't change anytime soon. I don't see another way for us to tell Vault to use the default value for these parameters on this specific API call, other than using the actual default value.